libnavigation-base/src/main/java/com/mapbox/navigation/base/route/model/RouteStepProgressNavigation.kt

```diff
@@ -0,0 +1,137 @@
+package com.mapbox.navigation.base.route.model
+
+data class RouteStepProgressNavigation internal constructor(
```

Comment:
It's not of much use making the constructor private, because even if it is not used from outside other than from the `Builder`, you can still change the data using the `copy` method. [Reference](http://gopalkri.com/2017/08/18/Kotlin-Data-Class-Constructor-Problem-Followup/)

Reply:
I have removed the `data` keyword and made the `class` `private`. Is this what you meant?

Reply:
`data class RouteStepProgressNavigation private constructor(`

I'm thinking about above to encapsulate better and force us to use a builder instead of the constructor when instantiating this object in the SDK.

Reply:
Adding `private` to `data` class doesn't make much sense. It is because you can still instantiate this class using the `copy` operator. I have rather converted them to regular `class` and made the constructor private. Hope it makes sense.